### PR TITLE
fix: #222 don't add extra columns in `step_population_scaling` or `layer_population_scaling`

### LINKED ISSUE #222: `step_population_scaling` includes all non-pop columns from the `df`
Unsure whether this is desired behavior or not. Here's an example created by using the built-in `state_census` and slightly modifying the example on the reference page.
The setup
```R
jhu <- epiprocess::jhu_csse_daily_subset %>% dplyr::select(geo_value, time_value, cases)
r <- epi_recipe(jhu) %>%
  step_population_scaling(df = state_census,
                          df_pop_col = "pop",
                          by = c("geo_value" = "abbr"),
                          cases, suffix = "_scaled") %>%
  step_epi_lag(cases_scaled, lag = c(0, 7, 14)) %>%
  step_epi_ahead(cases_scaled, ahead = 7, role = "outcome") %>%
  step_epi_naomit()
f <- frosting() %>%
  layer_predict() %>%
  layer_threshold(.pred) %>%
  layer_naomit(.pred) %>%
  layer_population_scaling(.pred, df = state_census,
                           by =  c("geo_value" = "abbr"),
                           df_pop_col = "pop")
wf <- epi_workflow(r, parsnip::linear_reg()) %>%
  fit(jhu) %>%
  add_frosting(f)
```
And here's the resulting transformation
```R
> summary(recipes::prep(r, jhu))
# A tibble: 10 × 4
   variable             type    role       source  
   <chr>                <chr>   <chr>      <chr>   
 1 time_value           date    time_value original
 2 geo_value            nominal geo_value  original
 3 cases                numeric raw        original
 4 fips                 numeric predictor  derived 
 5 name                 nominal predictor  derived 
 6 cases_scaled         numeric predictor  derived 
 7 lag_0_cases_scaled   numeric predictor  derived 
 8 lag_7_cases_scaled   numeric predictor  derived 
 9 lag_14_cases_scaled  numeric predictor  derived 
10 ahead_7_cases_scaled numeric outcome    derived 
```

As is, the user needs to drop every column in `df` except `df_pop_col`, or they'll get very strange behavior.

I've been told this is related to #212 and #163.

### PR DESCRIPTION
**Pull request by dsweber2**
fixes #222. Hopefully the tests are fairly self explanatory. All this had to do was remove any column in the auxillary `df` and not in the main data frame, instead of _just_ the population column.

### CODE REVIEW

**Comment by dsweber2** [comment]
In the process of fixing the broken test, I found out that `as.Date` when called on integers (e.g. `as.Date(12345)`) needs to have a start date defined for R versions older than 4.3.1. So I added a fix for that. 

In addition, `load_all` is broken by trying to add `quantile_reg` and `smooth_quantile_reg` after they've already been added, so I added some checks before the models are defined to see if they already have been.

File: R/layer_population_scaling.R [hunk 1]
```diff
@@ -156,9 +157,10 @@ slather.layer_population_scaling <-
     ) %>%
       dplyr::mutate(dplyr::across(
         dplyr::all_of(col_names),
-        ~.x * !!pop_col / object$rate_rescaling ,
-        .names = "{.col}{suffix}")) %>%
-     dplyr::select(-!!pop_col)
+        ~ .x * !!pop_col / object$rate_rescaling,
+        .names = "{.col}{suffix}"
+      )) %>%
+      dplyr::select(-any_of(col_to_remove))
```
Comment (dajmcdon) [inline R/layer_population_scaling.R:163]: I suspect that `any_of()` needs a package prefix: `dplyr::any_of()`. I don't think we've imported it elsewhere.

File: R/step_population_scaling.R [hunk 2]
```diff
@@ -211,8 +212,7 @@ bake.step_population_scaling <- function(object,
         ~.x * object$rate_rescaling /!!pop_col ,
         .names = "{.col}{suffix}")) %>%
     # removed so the models do not use the population column
-   dplyr::select(-!!pop_col)
-
+    dplyr::select(-any_of(col_to_remove))
```
Comment (dajmcdon) [inline R/step_population_scaling.R:215]: same as above.

File: R/utils-arg.R [hunk 3]
```diff
@@ -179,7 +179,7 @@ arg_is_sorted = function(..., allow_null = FALSE) {
 arg_to_date <- function(x, allow_null = FALSE, allow_na = FALSE) {
   arg_is_scalar(x, allow_null = allow_null, allow_na = allow_na)
   if (!is.null(x)) {
-    x <- tryCatch(as.Date(x), error = function(e) NA)
+    x <- tryCatch(as.Date(x, origin = "1970-01-01"), error = function(e) NA)
```
Comment (dajmcdon) [inline R/utils-arg.R:182]: Thanks!

File: R/zzz.R [hunk 4]
```diff
@@ -6,7 +6,10 @@
...
 .onLoad <- function(libname, pkgname) {
   make_flatline_reg()
-  make_quantile_reg()
-  make_smooth_quantile_reg()
+  if (!exists("quantile_reg")) {
+    make_quantile_reg()
+  }
+  if (!exists("smooth_quantile_reg")) {
+    make_smooth_quantile_reg()
+  }
```
Comment (dajmcdon) [inline R/zzz.R:14]: I haven't seen this in other packages that create parsnip models. This works? If yes, then we should delete the `.Rprofile` fix I added in #219

**Review by dajmcdon — CHANGES_REQUESTED** [review]

File: R/zzz.R (hunk 4 above)
Comment (dsweber2) [reply R/zzz.R:14]: I wish the `check_model_doesnt_exist` internal of parsnip were actually exposed for use, but this was working fine for me (I've been `load_all`'ing epipredict from other places, so I didn't notice the `.Rprofile`).

**Comment by dsweber2** [comment]
odd, looks like `if (!exists("quantile_reg"))` doesn't work in the github action, but does work on my local machine. I'll remove it. I'm also getting 3 failed tests
```
Failure (test-population_scaling.R:115:3): Postprocessing workflow works and values correct
`p <- predict(wf, latest)` produced warnings.
```
caused by a warning
```
Warning (test-population_scaling.R:224:3): test joining by default columns
prediction from a rank-deficient fit may be misleading
Backtrace:
     ▆
  1. ├─testthat::expect_message(p <- predict(wf, latest)) at test-population_scaling.R:224:2
  2. │ └─testthat:::expect_condition_matching(...)
  3. │   └─testthat:::quasi_capture(...)
  4. │     ├─testthat (local) .capture(...)
  5. │     │ └─base::withCallingHandlers(...)
  6. │     └─rlang::eval_bare(quo_get_expr(.quo), quo_get_env(.quo))
  7. ├─stats::predict(wf, latest)
  8. ├─epipredict:::predict.epi_workflow(wf, latest)
  9. │ ├─epipredict::apply_frosting(...) at epipredict/R/epi_workflow.R:123:2
 10. │ └─epipredict:::apply_frosting.epi_workflow(...) at epipredict/R/frosting.R:209:2
 11. │   ├─epipredict::slather(la, components, the_fit, the_recipe) at epipredict/R/frosting.R:263:6
 12. │   └─epipredict:::slather.layer_predict(la, components, the_fit, the_recipe) at epipredict/R/layers.R:134:2
 13. │     ├─stats::predict(...) at epipredict/R/layer_predict.R:67:2
 14. │     └─parsnip::predict.model_fit(...)
 15. │       ├─parsnip::predict_numeric(...) at parsnip/R/predict.R:155:2
 16. │       └─parsnip::predict_numeric.model_fit(...) at parsnip/R/predict_numeric.R:50:19
 17. │         └─rlang::eval_tidy(pred_call) at parsnip/R/predict_numeric.R:29:2
 18. ├─stats::predict(object = object$fit, newdata = new_data, type = "response")
 19. └─stats::predict.lm(object = object$fit, newdata = new_data, type = "response")
```
that it doesn't seem to be getting.

**Review by dajmcdon — APPROVED** [review]

**Comment by dsweber2** [comment]
I don't have write access so I can't actually merge this btw, if that's what you were intending post review. Thanks for the quick turn-around

**Pull request merged**